src/sagemaker/workflow/steps.py

```diff
@@ -148,6 +148,33 @@ def _resolve_depends_on(
                 raise ValueError(f"Invalid input step name: {step}")
         return depends_on
 
+    def _find_step_dependencies(self):
+        """Find the step names this step is dependent on."""
+
+        def _find_dependencies_in_step_arguments(obj):
+            dependencies = set()
+            if isinstance(obj, dict):
+                for value in obj.values():
+                    if isinstance(value, Properties):
+                        dependencies.add(value.step_name)
+                    dependencies.update(_find_dependencies_in_step_arguments(value))
+            if isinstance(obj, list):
+                for item in obj:
+                    if isinstance(item, Properties):
+                        dependencies.add(item.step_name)
+                    dependencies.update(_find_dependencies_in_step_arguments(item))
+            return dependencies
+
+        step_dependencies = set()
+        if self.depends_on:
+            for depends_on_step in self.depends_on:
+                if isinstance(depends_on_step, str):
+                    step_dependencies.add(depends_on_step)
+                elif isinstance(depends_on_step, Step):
+                    step_dependencies.add(depends_on_step.name)
```

Comment:
What if a StepCollection is in the depends on list?
This [merged commit](https://github.com/aws/sagemaker-python-sdk/commit/79ecab28a5004057c0842cfa291ef964d1f3129b#diff-f07f30c98bafb73739b39e93626762453e4249ce4f0b19c95692369397131378R141-R143) just supported the StepCollection to be depended on.
Can you add some tests on this as well?

Reply:
Right. The StepCollection needs to be treated with care.